analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.vb

```diff
@@ -0,0 +1,503 @@
+    End Function
+    Private Function IndependentConditions3(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return f.HasValue AndAlso d.Value = 42.0
+    End Function
+    Private Function IndependentConditions4(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return f Is Nothing AndAlso d.Value = 42.0
+    End Function
+    Private Function IndependentConditions5(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return f Is Nothing AndAlso d.Value = 42.0
+    End Function
+
+    Private Function DependentConditions1(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return Not d.HasValue AndAlso d.Value = 42.0                                    ' Noncompliant
+    End Function
+    Private Function DependentConditions2(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return d.Value = 42.0 AndAlso d.HasValue                                        ' Compliant
+    End Function
+    Private Function DependentConditions3(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return d.Value = 42.0 AndAlso d.HasValue = f.HasValue AndAlso f.Value = 42.0    ' Compliant
+    End Function
+    Private Function DependentConditions4(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return d.Value = 42.0 AndAlso d.HasValue <> f.HasValue AndAlso f.Value = 42.0   ' Noncompliant
+    End Function
+
+    Private Sub ThirdExcluded(ByVal b1 As Boolean?, ByVal b2 As Boolean?)
+        If b1 = b2 AndAlso b1 <> b2 Then
+            Dim x = (CTypeDynamic(Of Integer?)(Nothing)).Value                          ' FIXME Non-compliant
+        End If
+    End Sub
+
+    Private Sub Transitivity(ByVal b1 As Boolean?, ByVal b2 As Boolean?, ByVal b3 As Boolean?)
+        If b1 = b2 AndAlso b1 <> b3 AndAlso b2 = b3 Then
+            Dim x = (CTypeDynamic(Of Integer?)(Nothing)).Value                          ' FIXME Non-compliant
+        End If
+
+        If b1 <> b2 AndAlso b1 <> b3 AndAlso b2 <> b3 AndAlso b1 IsNot Nothing AndAlso b2 IsNot Nothing Then
+            Dim x = b3.Value                                                            ' FN: b3 is empty
+        End If
+    End Sub
+
+    Private Sub Reachability1()
+        Dim b1 As Boolean? = Nothing
+        Dim x = b1.Value                ' Noncompliant
+        x = b1.Value                    ' Compliant, unreachable
+    End Sub
+
+    Private Sub Reachability2(ByVal b1 As Boolean?)
+        Dim b2 As Boolean? = Nothing
+
+        If b1 = b2 Then
+            Dim x = b1.Value            ' Noncompliant
+            x = b2.Value                ' Noncompliant, FP: unreachable
+        End If
+    End Sub
+
+    Private Sub Reachability3(ByVal b1 As Boolean?, ByVal b2 As Boolean?)
+        Dim x = b1.Value Or b2.Value
+        x = b1.Value                    ' Compliant, both sides are evaluated
+        x = b2.Value                    ' Compliant, both sides are evaluated
+    End Sub
+End Class
+
+Class TernaryOperator
```

Comment:
It would be better to call the class `IfOperator`, since `If` is not necessarily ternary and acts as a replacement for other C# operators too (such as `??`).